### ISSUE TITLE
Question: is there a way to avoid duplicate labels in URL

### ISSUE THREAD

**Issue by bavarianbytes** [lochmueller/calendarize#641]
If a place the different views (list, month, day, details, etc.) on different pages (as required for all views except of "list + detail"), every page brings its own slug part. Furthermore the routes of Calendarize have a label in the routePath to identify the different views.
That leads to URLs like this:
```
domain.com/events/year/year/2011
domain.com/events/month/month/2011/11
domain.com/events/day/day/2011/11/17
``` 
The first part of the repeating words in the URL, like `month` results from the slug, the second `month` from the routePath of the routeEnhancers.
Is the any way to avoid that? As far as i know, you can't removed it from the slug as the parent page would have the same slug.

**Comment by okmiim** [comment]
Hi @bavarianbytes,
You should be able to omit the label if you use `limitToPages: [13]` with only one configuration per enhancer. And then repeat the plugin enhancer for each type (e.g., month, year, ...).
I currently cannot test that, so please share your results.

**Comment by bavarianbytes** [comment]
Thanks!
This is how it works for me now (almost):
```
routeEnhancers:

  CalendarizeDetail:
    type: Extbase
    extension: Calendarize
    plugin: Calendar
    limitToPages:
      - 183
    routes:
      - routePath: '/{index}'
        _controller: 'Calendar::detail'
    aspects:
      index:
        type: PersistedAliasMapper
        tableName: tx_calendarize_domain_model_index
        routeFieldName: slug

  CalendarizeYear:
    type: Extbase
    extension: Calendarize
    plugin: Calendar
    limitToPages:
      - 200
    routes:
      - routePath: '/{year}'
        _controller: 'Calendar::year'
    aspects:
      year:
        type: StaticRangeMapper
        start: '2015'
        end: '2040'

  CalendarizeQuarter:
    type: Extbase
    extension: Calendarize
    plugin: Calendar
    limitToPages:
      - 201
    routes:
      - routePath: '/{quarter}'
        _controller: 'Calendar::quarter'
    aspects:
      quarter:
        type: StaticRangeMapper
        start: '1'
        end: '4'

  CalendarizeMonth:
    type: Extbase
    extension: Calendarize
    plugin: Calendar
    limitToPages:
      - 184
    routes:
      - routePath: '/{year}/{month}'
        _controller: 'Calendar::month'
    aspects:
      year:
        type: StaticRangeMapper
        start: '2015'
        end: '2040'
      month:
        type: StaticRangeMapper
        start: '1'
        end: '12'

  CalendarizeWeek:
    type: Extbase
    extension: Calendarize
    plugin: Calendar
    limitToPages:
      - 202
    routes:
      - routePath: '/{year}/{week}'
        _controller: 'Calendar::week'
    aspects:
      year:
        type: StaticRangeMapper
        start: '2015'
        end: '2040'
      week:
        type: StaticRangeMapper
        start: '1'
        end: '53'

  CalendarizeDay:
    type: Extbase
    extension: Calendarize
    plugin: Calendar
    limitToPages:
      - 189
    routes:
      - routePath: '/{year}/{month}/{day}'
        _controller: 'Calendar::day'
    aspects:
      year:
        type: StaticRangeMapper
        start: '2015'
        end: '2040'
      month:
        type: StaticRangeMapper
        start: '1'
        end: '12'
      day:
        type: StaticRangeMapper
        start: '1'
        end: '31'

  CalendarizeOther:
    type: Extbase
    extension: Calendarize
    plugin: Calendar
    routes:
      - routePath: '/{calendarize_result_page_label}/{currentPage}'
        _controller: 'Calendar::result'
      - routePath: '/{calendarize_page_label}/{currentPage}'
        _controller: 'Calendar::list'
    aspects:
      currentPage:
        type: StaticRangeMapper
        start: '1'
        end: '1000'
      calendarize_result_page_label:
        type: LocaleModifier
        default: 'result-page'
        localeMap:
          - locale: 'de_.*'
            value: 'ergebnis-seite'
      calendarize_page_label:
        type: LocaleModifier
        default: 'page'
        localeMap:
          - locale: 'de_.*'
            value: 'seite'
```

The only thing that doesn't work for me is the paging of list and result view. I get links like `https://domain.com/termine/liste?tx_calendarize_calendar%5Bcontroller%5D=Calendar&tx_calendarize_calendar%5BcurrentPage%5D=2&cHash=1f817553d0dc96f982e1bfc0884b7b87` instead of speaking URLs.

But i realized that it doesn't has to do with my routeEnhancers. Even if import them from the calendarize extension and use the default templates, speaking URLs for paging don't work.

Is there a bug? Isn't there the `action` missing?

**Comment by okmiim** [comment]
Hi @bavarianbytes

In this case, the `action` is not missing. The `list` action is the [first one registered](https://github.com/lochmueller/calendarize/blob/b364af67a3aeb74fc6bdb9d2a56192b34c195991/ext_localconf.php#L21), so it is the default action and is generated without that parameter.

The 'non-speaking' URL has a different cause. For this URL, there is no matching route enhancer since your routes always have the parameter `{currentPage}`. This can be fixed in multiple ways:

- Add a default value for `currentPage`
  ```yaml
  routeEnhancers:
    CalendarizeOther:
      defaults:
        currentPage: 1
  ```
- Add a different route without `currentPage`
  ```yaml
  routeEnhancers:
    CalendarizeOther:
      routes:
        - routePath: '/{calendarize_page_label}'
          _controller: 'Calendar::list'
  ```

**Comment by bavarianbytes** [comment]
@okmiim for your fix #644!
Works great.
